src/mercury_engine_data_structures/formats/collision.py

```diff
@@ -91,26 +95,102 @@
+        """
+        Adds a new point by copying an existing point and inserting it at a specified index
+        param position: the x,y position of the new point
+        param idx: the index the new point will be placed in the poly
+        """
+        new_point = copy.deepcopy(self.get_point(0))
+        new_point.position = position
+        self._raw.points.insert(idx, new_point)
+        self.num_points += 1
+
+
+class PointData:
+    def __init__(self, raw: Container):
+        self._raw = raw
+
+    @property
+    def position(self) -> Vec2:
+        return self._raw.position
+
+    @position.setter
+    def position(self, value: Vec2) -> None:
+        self._raw.position = value
+
+    @property
+    def material_attribute(self) -> int:
+        return self._raw.material_attribute
+
+    @material_attribute.setter
+    def material_attribute(self, value: int = 1) -> None:
+        self._raw.material_attribute = value
+
+
 class CollisionEntry:
     def __init__(self, raw: Container):
         self._raw = raw
 
-    def get_data(self) -> Container:
-        """Returns all data of collision/collision_camera/logic shape"""
+    @property
+    def data(self) -> dict[Vec3, list[dict], Vec4, list[dict]]:
         return self._raw.data
 
-    def get_poly(self, poly_idx: int):
-        """Returns all data associated with a poly (points, boundings)"""
-        return self.get_data().polys[poly_idx]
+    @data.setter
+    def data(self, value: dict[Vec3, list[dict], Vec4, list[dict]]) -> None:
+        self._raw.data = value
+
+    @property
+    def position(self) -> Vec3:
+        return self.data.position
+
+    @position.setter
+    def position(self, value: Vec3) -> None:
+        self.data.position = value
+
+    @property
+    def total_boundings(self) -> Vec4:
+        return self.data.total_boundings
+
+    @total_boundings.setter
+    def total_boundings(self, value: Vec4) -> None:
+        self.data.total_boundings = value
+
+    @property
+    def polys(self) -> list[dict]:
+        return self.data.polys
 
-    def get_point(self, poly_idx: int, point_idx: int) -> Container:
-        """Returns a specific point in a poly"""
-        return self.get_poly(poly_idx).points[point_idx]
+    @polys.setter
+    def polys(self, value: list[dict]) -> None:
+        self.data.polys = value
 
-    def get_total_boundings(self) -> Container:
-        """Returns the total boundary of collision/collision_camera/logic shape"""
-        return self.get_data().total_boundings
+    def get_poly(self, poly_idx: int) -> PolyData:
+        """Returns all data associated with a poly"""
+        return PolyData(self.polys[poly_idx])
 
-    def get_poly_boundings(self, poly_idx: int) -> Container:
-        """Returns the boundary of a poly"""
-        return self.get_poly(poly_idx).boundings
+    def get_bst(self, bst_idx: int) -> Container:
+        """Returns a binary search tree"""
+        return self.data.binary_search_trees[bst_idx]
```

Comment:
similar question: do you need this for something in particular? I don't want to expose a Container if we don't need to

Reply:
No I was just being complete. I could remove this function.